internal/gcs/process.go

```diff
@@ -123,26 +127,53 @@ func (gc *GuestConnection) exec(ctx context.Context, cid string, params interfac
 
 // Close releases resources associated with the process and closes the
 // associated standard IO streams.
-func (p *Process) Close() error {
+func (p *Process) Close() (err error) {
 	ctx, span := oc.StartSpan(context.Background(), "gcs::Process::Close")
 	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
 	span.AddAttributes(
 		trace.StringAttribute("cid", p.cid),
 		trace.Int64Attribute("pid", int64(p.id)))
 
-	if err := p.stdin.Close(); err != nil {
-		log.G(ctx).WithError(err).Warn("close stdin failed")
+	if p.gcClosed() {
+		return hcs.ErrAlreadyClosed
 	}
-	if err := p.stdout.Close(); err != nil {
-		log.G(ctx).WithError(err).Warn("close stdout failed")
+
+	p.stdioLock.Lock()
+	if p.stdin != nil {
+		if err = p.stdin.Close(); err != nil {
+			log.G(ctx).WithError(err).Warn("close stdin failed")
+		}
+		p.stdin = nil
+	}
+	if p.stdout != nil {
+		if serr := p.stdout.Close(); serr != nil {
+			log.G(ctx).WithError(serr).Warn("close stdout failed")
+			if err == nil {
+				err = serr
+			}
+		}
+		p.stdout = nil
 	}
-	if err := p.stderr.Close(); err != nil {
-		log.G(ctx).WithError(err).Warn("close stderr failed")
+	if p.stderr != nil {
+		if serr := p.stderr.Close(); serr != nil {
+			log.G(ctx).WithError(serr).Warn("close stderr failed")
+			if err == nil {
+				err = serr
```

Comment:
We don't actually do anything with these assignments right? After all of these close, whether successful or not we just return nil on line 173.

Reply:
Or is it some weird Go named return value shenanigans that I'm forgetting

Reply:
We use it to set the span status, but I dont know if we should even bother to return stdIO close errors (or log it in the span).
`hcs\process.Close` just ignores stdIO errors.
We probably should standardize how we do closes ....

Reply:
Ahh, for the span okay. And yes agreed. It's frustrating to go look at how each type of container we support does something, and they all have their own little quirks or way of handling something

Reply:
When the function returns though, that value gets set to the "err" value of the function. So then the span would still be getting set to nil error. Right?

Reply:
oh right, i missed that ...